src/ui/src/builder/journal/BuilderJournalBlockOutput.vue

```diff
@@ -0,0 +1,449 @@
+				v-if="output.stdout && activeTab === 'stdout'"
+				class="output-block__log-section"
+			>
+				<pre class="output-block__log-content">{{ output.stdout }}</pre>
+			</div>
+
+			<!-- Logs Tab -->
+			<div
+				v-if="output.logs && activeTab === 'logs'"
+				class="output-block__log-section"
+			>
+				<pre class="output-block__log-content">{{ output.logs }}</pre>
+			</div>
+
+			<!-- Result Tab -->
+			<div
+				v-if="activeTab === 'result'"
+				class="output-block__result-section"
+			>
+				<SharedJsonViewer
+					:data="output.result"
+					is-root
+					is-root-open
+					enable-copy-to-json
+				/>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script setup lang="ts">
+import { computed, PropType, ref, watch } from "vue";
+import WdsIcon from "@/wds/WdsIcon.vue";
+import WdsTabs from "@/wds/WdsTabs.vue";
+import SharedImgWithFallback from "@/components/shared/SharedImgWithFallback.vue";
+import SharedJsonViewer from "@/components/shared/SharedJsonViewer/SharedJsonViewer.vue";
+import { convertAbsolutePathtoFullURL } from "@/utils/url";
+
+type BlockOutput = {
+	result: any;
+	outcome: string;
+	message?: string;
+	stdout?: string;
+	logs?: string;
+	startedAt?: number;
+	executionTimeInSeconds?: number;
+	component?: {
+		type?: string;
+		title?: string;
+	} | null;
+};
+
+const props = defineProps({
+	blockId: { type: String, required: true },
+	output: { type: Object as PropType<BlockOutput>, required: true },
+	index: { type: Number, required: true },
+});
+
+defineEmits<{
+	"go-to-block": [blockId: string];
+}>();
+
+const activeTab = ref<string>("");
+
+// Build tabs based on available data
+const blockTabs = computed(() => {
+	const tabs = [];
+
+	if (props.output.message) {
+		tabs.push({ label: "Error", value: "error" });
+	}
+	if (props.output.stdout) {
+		tabs.push({ label: "Stdout", value: "stdout" });
+	}
+	if (props.output.logs) {
+		tabs.push({ label: "Logs", value: "logs" });
+	}
+	tabs.push({ label: "Result", value: "result" });
+
+	return tabs;
+});
+
+watch(
+	blockTabs,
+	(tabs) => {
+		if (!tabs.length) {
+			activeTab.value = "";
+			return;
+		}
+		if (!tabs.some((tab) => tab.value === activeTab.value)) {
+			activeTab.value = tabs[0].value;
+		}
+	},
+	{ immediate: true },
+);
+
+function formatTimestamp(timestamp: number): string {
+	const date = new Date(timestamp * 1000);
+	return date.toLocaleTimeString("en-US", {
+		hour: "2-digit",
+		minute: "2-digit",
+		second: "2-digit",
+		hour12: true,
+	});
+}
+
+function formatDuration(seconds: number): string {
+	if (seconds < 1) {
+		return `${(seconds * 1000).toFixed(0)}ms`;
+	}
+	return `${seconds.toFixed(2)}s`;
+}
+
+function getOutcomeClass(outcome: string): string {
+	const outcomeMap: Record<string, string> = {
+		success: "output-block--success",
+		trigger: "output-block--trigger",
+		skipped: "output-block--skipped",
+		error: "output-block--error",
+		stopped: "output-block--skipped",
+	};
+	return outcomeMap[outcome] || "output-block--success";
+}
```

Comment:
It's better to avoid creating an object on each change, and it needs to be a computed

```suggestion
const rootClassByOutcome: Record<string, string> = {
		success: "output-block--success",
		trigger: "output-block--trigger",
		skipped: "output-block--skipped",
		error: "output-block--error",
		stopped: "output-block--skipped",
	};

const rootClass = computed(() =>rootClassByOutcome[props.outcome] || "output-block--success")
```